### PR TITLE
Added Initial tables

### PR DESCRIPTION
**Pull request by bigdatasourav**

### CODE REVIEW

File: docs/tables/sentry_key.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Table: sentry_key
+
+Represent a list of client keys bound to a project.
```
Comment (misraved) [inline docs/tables/sentry_key.md:3]:
```suggestion
Represents a list of client keys bound to a project.
```

File: docs/tables/sentry_organization.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Table: sentry_organization
+
+Represent a list of organizations available to the authenticated session.
```
Comment (misraved) [inline docs/tables/sentry_organization.md:3]:
```suggestion
Represents a list of organizations available to the authenticated session.
```

File: docs/tables/sentry_organization.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  status ->> 'id' <> 'active';
+```
+
+### List access details of a particular organization
+
+```sql
+select
+  id,
+  name,
+  jsonb_pretty(access) as access
```
Comment (misraved) [inline docs/tables/sentry_organization.md:91]: In this example query, would it be better to list out the organization's access details by opening up the `access` column?

File: docs/tables/sentry_organization_member.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
+# Table: sentry_organization_member
+
+Represent a list of users that belong to a given organization.
```
Comment (misraved) [inline docs/tables/sentry_organization_member.md:3]:
```suggestion
Represents a list of users that belong to a particular organization.
```

File: docs/tables/sentry_organization_repository.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# Table: sentry_organization_repository
+
+Represent a list of version control repositories for a given organization.
```
Comment (misraved) [inline docs/tables/sentry_organization_repository.md:3]:
```suggestion
Represents a list of version control repositories for a given organization.
```

File: sentry/table_sentry_issue_alert.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+		} else {
+			break
+		}
+	}
+
+	return nil, nil
+}
+
+func getIssueAlert(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	orgSlug := d.EqualsQuals["organization_slug"].GetStringValue()
```
Comment (misraved) [inline sentry/table_sentry_issue_alert.go:173]:
```suggestion
	orgSlug := d.EqualsQualString["organization_slug"]
```
@bigdatasourav I believe we could make this change across all the qual references.

File: sentry/table_sentry_key.go [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+			},
+			{
+				Name:        "label",
+				Type:        proto.ColumnType_STRING,
+				Description: "The label of the key.",
+			},
+			{
+				Name:        "project_id",
+				Type:        proto.ColumnType_INT,
+				Description: "The ID of the project the keys belong to.",
```
Comment (misraved) [inline sentry/table_sentry_key.go:66]:
```suggestion
				Description: "The ID of the project the key belongs to.",
```

File: sentry/table_sentry_metric_alert.go [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+			},
+			{
+				Name:        "name",
+				Type:        proto.ColumnType_STRING,
+				Description: "The metric alert name.",
+			},
+			{
+				Name:        "owner",
+				Type:        proto.ColumnType_STRING,
+				Description: "Specifies the owner id of this metric alert rule.",
```
Comment (misraved) [inline sentry/table_sentry_metric_alert.go:45]:
```suggestion
				Description: "Specifies the owner ID of this metric alert rule.",
```

File: sentry/table_sentry_metric_alert.go [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+			},
+			{
+				Name:        "task_uuid",
+				Type:        proto.ColumnType_STRING,
+				Description: "The UUID of the async task that can be spawned to create the metric alert.",
+				Transform:   transform.FromField("TaskUUID"),
+			},
+			{
+				Name:        "threshold_type",
+				Type:        proto.ColumnType_INT,
```
Comment (misraved) [inline sentry/table_sentry_metric_alert.go:95]: Is the data type of the column correct?

File: sentry/table_sentry_organization.go [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+			},
+			{
+				Name:        "relay_pii_config",
+				Type:        proto.ColumnType_STRING,
+				Description: "The relay pii config.",
+			},
+			{
+				Name:        "require_email_verification",
+				Type:        proto.ColumnType_BOOL,
+				Description: "Check if the organization require email verification.",
```
Comment (misraved) [inline sentry/table_sentry_organization.go:134]:
```suggestion
				Description: "Check if the organization requires email verification.",
```

File: sentry/table_sentry_organization.go [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+			},
+			{
+				Name:        "scrape_java_script",
+				Type:        proto.ColumnType_BOOL,
+				Description: "Check if the organization scrape java script.",
```
Comment (misraved) [inline sentry/table_sentry_organization.go:139]:
```suggestion
				Description: "Check if the organization scrapes java script.",
```

File: sentry/table_sentry_organization.go [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+			},
+			{
+				Name:        "scrub_ip_addresses",
+				Type:        proto.ColumnType_BOOL,
+				Description: "Check if the organization scrub IP addresses.",
```
Comment (misraved) [inline sentry/table_sentry_organization.go:144]:
```suggestion
				Description: "Check if the organization scrubs IP addresses.",
```

File: sentry/table_sentry_project.go [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+func listProjects(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	conn, err := getClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("listProjects", "connection_error", err)
```
Comment (misraved) [inline sentry/table_sentry_project.go:235]:
```suggestion
		plugin.Logger(ctx).Error("sentry_project.listProjects", "connection_error", err)
```
@bigdatasourav could you please append the table name to the error logs?

**Review by misraved — COMMENTED** [review]
@bigdatasourav please take a look at the initial set of review comments. Thanks!!

Also, in e example queries could you please use a made-up organization_slug? It is better to avoid names like `test`.

File: .DS_Store
[the remark this answers is not in the document]
Comment (bigdatasourav) [reply .DS_Store:1]: Removed

File: docs/index.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
+---
+organization: Turbot
+category: ["software development"]
+icon_url: "/images/plugins/turbot/sentry.svg"
+brand_color: "#362d59"
```
[the remark this answers is not in the document]
Comment (bigdatasourav) [reply docs/index.md]: Updated

File: config/sentry.spc [hunk 15]
```diff
@@ -0,0 +1,10 @@
+connection "sentry" {
+  plugin = "sentry"
+
+  # `auth_token` - Create an authentication token in the Sentry Console for use.
+  # Console path - settings -> account -> api -> auth-tokens
```
[the remark this answers is not in the document]
Comment (bigdatasourav) [reply config/sentry.spc]: Updated the link as -  # For more information on the Auth Token, please see https://docs.sentry.io/api/auth/.

File: docs/index.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
+---
+organization: Turbot
+category: ["software development"]
```
[the remark this answers is not in the document]
Comment (bigdatasourav) [reply docs/index.md:3]: It is a developer-first error tracking and performance monitoring platform, so software development will be the right category.

File: docs/tables/sentry_organization.md (hunk 3 above)
Comment (bigdatasourav) [reply docs/tables/sentry_organization.md:91]: In API response, it is an array of string, https://pkg.go.dev/github.com/jianyuan/go-sentry/v2/sentry#Organization.Access

File: sentry/table_sentry_issue_alert.go (hunk 6 above)
Comment (bigdatasourav) [reply sentry/table_sentry_issue_alert.go:173]: Updated with d.EqualsQualString()

File: sentry/table_sentry_key.go (hunk 7 above)
Comment (bigdatasourav) [reply sentry/table_sentry_key.go:66]: updated

File: sentry/table_sentry_metric_alert.go (hunk 8 above)
Comment (bigdatasourav) [reply sentry/table_sentry_metric_alert.go:45]: Updated

File: sentry/table_sentry_metric_alert.go (hunk 9 above)
Comment (bigdatasourav) [reply sentry/table_sentry_metric_alert.go:95]: yes, https://pkg.go.dev/github.com/jianyuan/go-sentry/v2/sentry#MetricAlert.ThresholdType

File: sentry/table_sentry_organization.go (hunk 10 above)
Comment (bigdatasourav) [reply sentry/table_sentry_organization.go:134]: updated

File: sentry/table_sentry_organization.go (hunk 11 above)
Comment (bigdatasourav) [reply sentry/table_sentry_organization.go:139]: updated

File: sentry/table_sentry_organization.go (hunk 12 above)
Comment (bigdatasourav) [reply sentry/table_sentry_organization.go:144]: updated

File: sentry/table_sentry_project.go (hunk 13 above)
Comment (bigdatasourav) [reply sentry/table_sentry_project.go:235]: updated

**Review by misraved — APPROVED** [review]

**Pull request merged**